crypto/decrypting_reader.go

```diff
@@ -0,0 +1,134 @@
+package crypto
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/cipher"
+	"errors"
+	"io"
+)
+
+func newBlockModeDecryptingReader(r io.Reader, mode cipher.BlockMode) io.Reader {
+	return &blockModeDecryptingReader{
+		r:         bufio.NewReader(r),
+		blockMode: mode,
+		buffer:    bytes.NewBuffer(nil),
+		err:       nil,
+	}
+}
+
+type blockModeDecryptingReader struct {
+	r         *bufio.Reader
+	blockMode cipher.BlockMode
+	buffer    *bytes.Buffer
+	err       error
+}
+
+func (decryptingReader *blockModeDecryptingReader) readNextBlock() ([]byte, error) {
+	reader := decryptingReader.r
+
+	output := make([]byte, decryptingReader.blockMode.BlockSize())
+	sizeOfCurrentlyRead, readErr := io.ReadFull(reader, output)
+	if readErr != nil && readErr != io.EOF {
+		return output[:sizeOfCurrentlyRead], readErr
+	}
```

Comment:
Should we return any output in case of error?

Reply:
Probably not. I will remove it and if there's any problem report back to you to figure out what to do